### PR TITLE
#30 Adding more copy options on the row

### PR DESCRIPTION
**Pull request by rjmasikome**
So I added the more options on copy. I think from the UX capability it's probably fine to do it in two clicks. As it is also not clear what they want to copy just from one click in my opinion. I changed the icon to triple dots, as for the usability perspective it should mean there should be more options after clicking. But if you guys are not up for it, just changed for another icon.

Anyway if anything to change lemme know.

Cheers,
RJ

### CODE REVIEW

**Comment by weeco** [comment]
Fixes #30

File: frontend/src/components/pages/topics/Tab.Messages.tsx [hunk 1]
```diff
@@ -405,14 +424,19 @@ export class TopicMessageView extends Component<{ topic: TopicDetail }> {
                 onFilterDropdownVisibleChange: (_) => this.showColumnSettings = true,
                 filterIcon: (_) => {
                     return <Tooltip title='Column Settings' mouseEnterDelay={0.1}>
-                        <SettingFilled />
+                        <SettingFilled style={IsColumnSettingsEnabled ? { color: '#1890ff' } : { color : '#a092a0' }}/>
                     </Tooltip>
                 },
                 render: (text, record) => !record.isValueNull && (
                     <span>
                         <ZeroSizeWrapper width={32} height={0}>
```
Comment (rikimaru0345) [inline frontend/src/components/pages/topics/Tab.Messages.tsx:432]: 1. Icon is not vertically centered (but should be):
   


2. Clickable region of the button should fill the whole table-cell; remove padding in those (msgTableActionColumn) cells

    green: currently clickable
    red: should be clickable

Comment (rikimaru0345) [reply frontend/src/components/pages/topics/Tab.Messages.tsx:432]: I'd probably do something like this:
- remove usage of `<ZeroSizeWrapper>` here
- `.ant-table.ant-table-middle .ant-table-tbody>tr>td.msgTableActionColumn { padding: 0 }`
- ensure button fills the whole parent
- if the alignment of the icon inside the button is an issue, you can try making the button a flex container: something like `<Button style={{display:'flex', placeContent:'center'}} ...` or so

File: frontend/src/index.css [hunk 2]
```diff
@@ -663,18 +667,25 @@ tr:hover button.ant-btn-link {
...
 .ant-table-middle>.ant-table-content>.ant-table-body>table>.ant-table-tbody>tr>td.msgTableActionColumn {
     padding-left: 0;
 }
 
-tr>td.msgTableActionColumn {
+.ant-table.ant-table-middle .ant-table-tbody>tr>td.msgTableActionColumn {
+    padding-bottom: 0;
+}
+
+tr>td {
```
[the remark this answers is not in the document]
Comment (rikimaru0345) [reply frontend/src/index.css:682]: actually, I just remembered that neither solution is necessary 😄 😅 
we can probably just set `align:'right'` in the column definitions (where column name, data index etc is set)!!

**Pull request merged**